c/src/ml-api-service-offloading.c

```diff
@@ -387,24 +403,42 @@ _mlrs_process_service_offloading (nns_edge_data_h data_h, void *user_data)
         "Failed to get service key while processing the ml-offloading service.");
   }
 
+  dir_path = _mlrs_get_model_dir_path (offloading_s, service_key);
+
+  if (offloading_s->offloading_mode == ML_SERVICE_OFFLOADING_MODE_TYPE_TRAINING) {
+    ml_service_training_offloading_process_received_data (mls, data_h, dir_path,
+        data, service_type);
+    if (service_type == ML_SERVICE_OFFLOADING_TYPE_REPLY) {
+      if (!dir_path) {
+        _ml_error_report_return (NNS_EDGE_ERROR_UNKNOWN,
+            "Failed to get model directory path.");
+      }
+
+      if (_mlrs_model_register (service_key, data_h, data, data_len, dir_path)) {
+        event_type = ML_SERVICE_EVENT_MODEL_REGISTERED;
+      } else {
+        _ml_error_report ("Failed to register model downloaded from: %s.",
+            (gchar *) data);
+      }
+    }
+  }
+
   switch (service_type) {
     case ML_SERVICE_OFFLOADING_TYPE_MODEL_URI:
     {
       GByteArray *array = g_byte_array_new ();
-      g_autofree gchar *dir_path = NULL;
+
+      if (!dir_path) {
```

Comment:
Is it required to move this check to 407, which is just after assigning ```dir_path```?

Reply:
ML_SERVICE_OFFLOADING_TYPE_PIPELINE_URI, ML_SERVICE_OFFLOADING_TYPE_PIPELINE_RAW and ML_SERVICE_OFFLOADING_TYPE_REPLY do not use dir_path.